### PR TITLE
Return group needsRotation to admins

### PR DESCRIPTION
**Pull request by giarc3**
- Added `needsRotation` as an `Option[Boolean]` to `GroupMetaResult`
- Added `GroupGetResult` and `groupGetMetadata()`
- Added test for `needsRotation` on `groupCreate()`
- Added test for `groupGetMetadata()`

TODO:
- [x] Depend on released ironoxide-scala

### CODE REVIEW

File: src/main/scala/com/ironcorelabs/GroupGetResult.scala [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+/**
+ * Group information
+ */
+case class GroupGetResult(
+  id: GroupId,
+  name: Option[GroupName],
+  groupMasterPublicKey: PublicKey,
+  isAdmin: Boolean,
+  isMember: Boolean,
+  adminList: Option[List[UserId]],
```
Comment (coltfred) [inline src/main/scala/com/ironcorelabs/GroupGetResult.scala]: We should add a comment about the difference between None and Some(Nil).

File: src/test/scala/ironoxide/FullIntegrationTest.scala [hunk 2]
```diff
@@ -62,16 +63,37 @@ class FullIntegrationTest extends AsyncWordSpec with Matchers with EitherValues
...
+  "Group Get" should {
+    "Return data for valid group admin" in {
+      val sdk = IronSdkSync[IO](deviceContext)
+      val groupGetResult = sdk.groupGetMetadata(validGroupId).attempt.unsafeRunSync.value
+
+      groupGetResult.id.id.length shouldBe 36
+      groupGetResult.name.get.name shouldBe "a name"
+      groupGetResult.isAdmin shouldBe true
+      groupGetResult.isMember shouldBe true
+      groupGetResult.adminList shouldBe Some(List(primaryTestUserId))
```
Comment (coltfred) [inline src/test/scala/ironoxide/FullIntegrationTest.scala]: I think you can use `.value` on admin list to not assert about `Some(...)`

File: src/main/scala/com/ironcorelabs/GroupGetResult.scala (hunk 1 above)
Comment (giarc3) [reply src/main/scala/com/ironcorelabs/GroupGetResult.scala]: Added comments to all parameters

File: src/main/scala/com/ironcorelabs/GroupGetResult.scala [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  created: ju.Date,
+  lastUpdated: ju.Date,
+  needsRotation: Option[Boolean]
+)
+
+object GroupGetResult {
+  def apply(gmr: jsdk.GroupGetResult): GroupGetResult = {
+    val optName: Option[GroupName] = gmr.getName.toScala.map(n => GroupName(n.getName))
+    val optRotation = gmr.getNeedsRotation.toScala.map(_.getBoolean)
+    val optAdmins = gmr.getAdminList.toScala.map(_.getList.toList.map(id => UserId(id)))
```
[the remark this answers is not in the document]
Comment (giarc3) [reply src/main/scala/com/ironcorelabs/GroupGetResult.scala]: Done

File: src/test/scala/ironoxide/FullIntegrationTest.scala (hunk 2 above)
Comment (giarc3) [reply src/test/scala/ironoxide/FullIntegrationTest.scala]: Done

**Pull request merged**